app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt

```diff
@@ -285,75 +309,128 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
+                    if (classificationView == null) {
+                        return@forEach
+                    }
 
-        if (chips.isEmpty()) {
+                    classificationView.setSelectable(true)
 
-            formChipsLabel.visibility = View.GONE
-            formChipsViews.visibility = View.GONE
-            input.setHint(getString(R.string.feedback_form_hint))
+                    val index = classificationView.tag as Int
 
-        } else {
+                    if (model.closedSet[index].toBoolean()) {
+                        classificationView.addChip(classificationView.getChips().size, FeedbackChip.createOther(OTHER_TITLE))
+                    }
 
-            formChipsViews.setChips(chips)
+                    classificationView.setSelectCallback {
 
-            formChipsViews.setSelectCallback { _ ->
-                disableInput()
-                enableConfirm()
-            }
+                        if ((it as FeedbackChip).isOther()) {
+                            showNegativeFormInput(index, it, classificationView)
+                        }
 
-            formChipsViews.setDeselectCallback { _ ->
-                enableInput()
-                disableConfirm()
-            }
-        }
+                        currentClassifications.put(index, it)
+                        if (currentClassifications != initialClassifications && currentClassifications.size == initialClassifications.size) {
+                            enableNegativeFormConfirm()
+                        } else {
+                            disableNegativeFormConfirm()
+                        }
+                    }
 
-        feedbackView.visibility = View.GONE
-        feedbackFormView.visibility = View.VISIBLE
+                    classificationView.setDeselectCallback {
+                        currentClassifications.remove(classificationView.tag as Int)
+                        disableNegativeFormConfirm()
+                    }
+                }
+
+        toggleContainer.visibility = View.GONE
+        negativeFormButtonsContainer.visibility = View.VISIBLE
+
+        expandResults()
+    }
+
+    private fun enableNegativeFormConfirm() {
+        negativeFormConfirmButton.isEnabled = true
+    }
+
+    private fun disableNegativeFormConfirm() {
+        negativeFormConfirmButton.isEnabled = false
+    }
+
+    private fun hideNegativeForm(reset: Boolean = true) {
+        positiveButton.visibility = View.VISIBLE
+        negativeButton.isEnabled = true
+
+        (0 until otherPredictionsContainer.childCount)
+                .map { otherPredictionsContainer.getChildAt(it) as? ClassificationsView }
+                .forEach {
+                    it?.setSelectable(false)
+                    if (reset) {
+                        it?.deselectAll()
+                        it?.select(0)
+
+                        if (model.closedSet[it?.tag as Int].toBoolean()) {
+                            it.removeChip(it.getChips().size - 1)
```

Comment:
Won't this remove items that aren't "custom"?

Reply:
Changed